lib/ikea.js

```diff
@@ -17,6 +19,23 @@ async function bulbOnEvent(type, data, device) {
     }
 }
 
+async function batteryDeviceOnEvent(type, data, device) {
+    /**
+     * IKEA battery deivces sometimes lose the bind and
+     * reporting configuration after replacing the battery.
+     * A deviceAnnounce indicates they are powered on again.
+     * Reconfigure the bind and reporting here.
+     */
+    if (type === 'deviceAnnounce') {
+        // XXX: how do we get coordinatorEndpoint ?
+        const coordinatorEndpoint = ?; // this.zigbee.getDevicesByType('Coordinator')[0].getEndpoint(1)
+        const endpoint = device.getEndpoint(1);
+        await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
```

Comment:
Something like this should do it:
```js
for (const endpoint of device.endpoints) {
  const b = endpoint.binds.find(b => b.cluster.name === 'genPowerCfg');
  if (b) {
     await endpoint.bind('genPowerCfg', b.target);
  }

 const r = endpoint.configuredReportings.find(r => r.cluster.name === 'genPowerCfg' && r.attribute.name === 'batteryPercentageRemaining')
 if (r) {
    await endpoint.configureReporting(c.cluster.name, [{
            attribute: r.attribute.name, minimumReportInterval: r.minimumReportInterval,
            maximumReportInterval: r.maximumReportInterval, reportableChange: r.reportableChange,
        }]);
  }

}
```

Reply:
That seems to do the trick after making lint happy and undoing a lint change that broke it again.